shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -30,56 +32,214 @@ function _getLocalStorageObject(): Storage {
+                                }
+                            }
+                        },
+                        key: function(index: number): string {
+                            try {
+                                return originalStorage.key(index);
+                            } catch (e) {
+                                // Reset the cached storage on error
+                                if (storageType === StorageType.LocalStorage) {
+                                    _verifiedLocalStorage = null;
+                                } else {
+                                    _verifiedSessionStorage = null;
+                                }
+                                return null;
+                            }
+                        },
+                        get length(): number {
+                            try {
+                                return originalStorage.length;
+                            } catch (e) {
+                                // Reset the cached storage on error
+                                if (storageType === StorageType.LocalStorage) {
+                                    _verifiedLocalStorage = null;
+                                } else {
+                                    _verifiedSessionStorage = null;
+                                }
+                                return 0;
+                            }
+                        }
+                    };
+                    
+                    result = wrappedStorage as Storage;
+                }
+            } catch (e) {
+                // Storage exists but can't be used (quota exceeded, etc.)
+            }
         }
-    } catch (exception) {
-        // eslint-disable-next-line no-empty
+    } catch (e) {
+        // Catch any unexpected errors
     }
-
-    return null;
+    
+    return result;
 }
 
 /**
  * Gets the sessionStorage object if available
  * @returns {Storage} - Returns the storage object if available else returns null
  */
 function _getSessionStorageObject(): Storage {
-    if (utlCanUseSessionStorage()) {
-        return _getVerifiedStorageObject(StorageType.SessionStorage);
+    if (!_verifiedSessionStorage) {
+        _verifiedSessionStorage = createCachedValue(_getVerifiedStorageObject(StorageType.SessionStorage));
     }
+    return _verifiedSessionStorage.v;
+}
 
-    return null;
+/**
+ * Helper function to safely wrap storage operations in try/catch
+ * and reset cache if an operation fails
+ * @param storageFn - Function that performs a storage operation
+ * @param storageType - Type of storage being accessed
+ * @param fallbackValue - Value to return if operation fails
+ * @param logger - Optional logger for error reporting
+ * @returns Result of the storage operation or the fallback value
+ */
+function _safeStorageOperation<T>(storageFn: () => T, storageType: StorageType, fallbackValue: T, logger?: IDiagnosticLogger, errorMessageId?: _eInternalMessageId, errorMessage?: string): T {
```

Comment:
This looks like it's just added additional code -- I believe it can be removed to keep the generated code smaller.